christin/index.js

```diff
@@ -0,0 +1,237 @@
+
+*/
+
+const programming = {
+  languages: ["JavaScript", "Python", "Ruby"],
+  isChallenging: true,
+  isRewarding: true,
+  difficulty: 8,
+  jokes:
+    "http://stackoverflow.com/questions/234075/what-is-your-best-programmer-joke",
+};
+
+//a)
+
+function modifyProgrammingLanguage(data) {
+  programming.languages.push(data);
+  return programming;
+}
+
+console.log(modifyProgrammingLanguage("Go"));
+//------------------------------------------------
+
+// b)
+
+const modifyProgrammingData = (data) => {
+  if (data.language) {
+    programming.language = data.language;
+  }
+  if (data.isChallenging) {
+    programming.isChallenging0data.isChallenging;
+  }
+  if (data.isRewarding) {
+    programming.isRewarding = data.isRewarding;
+  }
+  if (data.difficulty) {
+    programming.difficulty = data.difficulty;
+  }
+  if (data.jokes) {
+    programming.joke = data.jokes;
+  }
+};
+
+modifyProgrammingData({
+  difficulty: 7,
+});
+
+console.log(programming);
+
+// c) - Using the delete keyword, write the command to remove the jokes key from the programming object.
+
+delete programming.jokes;
+
+//d) - Write a command to add a new key called isFun and a value of true to the programming object.
+
+programming.isFun = true;
+console.log(programming);
+
+//e) - Using a loop, iterate through the languages array and console.log all of the languages.
+
+let languages = [];
+
+function getLanguages(obj) {
+  for (let key in obj) {
+    if (key === "languages") {
+      languages = obj[key];
+    }
+  }
+  return languages;
+}
+
+getLanguages(programming);
+console.log(languages); // [ 'JavaScript', 'Python', 'Ruby', 'Go' ]
+
+//f) - Using a loop, console.log all of the keys in the programming object.
+
+keys = [];
+
+function getKeys(obj) {
+  for (let key in obj) {
+    keys.push(key);
+  }
+}
+
+getKeys(programming);
+console.log("keys=>", keys); // keys=> [ 'languages', 'isChallenging', 'isRewarding', 'difficulty', 'isFun' ]
+
+//g) - Using a loop, console.log all of the values in the programming object.
+
+let values = [];
+
+function getValues(obj) {
+  for (let key in obj) {
+    values.push(obj[key]);
+  }
+}
+
+getValues(programming);
+console.log("Values==>", values); // Values==> [ [ 'JavaScript', 'Python', 'Ruby', 'Go' ], true, true, 7, true ]
+
+//h) - Create an object method where if the keys "isChallenging" and "isRewarding" have values of "true", then return "Learning the programming languages: "JavaScript, Python, Ruby, Go" is rewarding and challenging. Bonus: In a comment, explain what is printed if we console.log an object method without calling it and why.*/
+
+programming.content = `Learning the programming Languages: "${languages}" is rewarding and challenging`;
+console.log(programming);
+
+if (programming.isChallenging === true && programming.isRewarding === true) {
+  console.log(programming.content);
+}
+
+// Bonus: In a comment, explain what is printed if we console.log an object method without calling it and why.*/
```

Comment:
Bonus ==> Make sure that any other code cannot delete or change properties of the object.
So we could use the Object.seal() method seals an object, preventing new properties from being added to it and marking all existing properties as non-configurable. Values of present properties can still be changed as long as they are writable. After using seal method if we try to delete languages we cant.
Object.seal(programming);
delete programming.languages;
console.log(programming)

We could also use freeze Object method instead of seal

Reply:
Hey, thanks thats interesting,
questions (maybe stupid ones):
- what you mean with writeable?
- in witch cases do you use seal? can you undo it for example in a software or something? 


and is this really the answer to the question:
/ Bonus: In a comment, explain what is printed if we console.log an object method without calling it and why.*/